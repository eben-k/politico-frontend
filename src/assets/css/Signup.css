.signup-form{
  margin: 2px auto 0 auto;
  width: 400px;
  height: 90vh;
  box-shadow: 1px 1px 1px 1px #9b8b85;
  padding: 10px 0 0 0;
  background: white;
}
.signup {
  margin: 5% auto 15% 5%;
  width: 100%;
 }
 input[type="text"], input[type="password"], input[type="tel"], input[type="number"]{
  width: 341px;
  height: 50px;
  margin-bottom: 26px;
  border: none;
  background: #f3efef;
  padding-left: 15px;
  outline: none;
}
input[type="submit"] {
  width: 350px;
  height: 50px;
  border: none;
  color: white;
  background: #7ecddd;
  cursor: pointer;
  font-size: 20px;
}
