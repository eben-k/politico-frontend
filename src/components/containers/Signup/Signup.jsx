import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { signupUser } from '../../../store/modules/auth';

import Layout from '../../presentationals/Layout';

import '../../../assets/css/Signup.css';

export const Signup = ({ signupUser: handleSignup }) => {
  const [userDetails, setUserDetails] = useState({
    firstname: '',
    lastname: '',
    email: '',
    passportUrl: '',
    phoneNumber: '',
    password: '',
  });

  function updateLocalState(event) {
    setUserDetails({ ...userDetails, [event.target.name]: event.target.value });
  }
  async function onSubmit(event) {
    event.preventDefault();
    const signupResponse = await handleSignup(userDetails);
    return signupResponse;
  }
  return (
    <Layout>
      <div className="signup-form">
        <h1 className="login-heading">Sign Up To Politico</h1>
        <div className="form">
          <form onSubmit={onSubmit} id="form" className="signup" action="">
            <div>
              <input
                id="firstname"
                name="firstname"
                type="text"
                placeholder="First Name"
                onChange={updateLocalState}
              />
              <div id="firstnameError" className="error" />
            </div>
            <div>
              <input
                id="lastname"
                name="lastname"
                type="text"
                placeholder="Last Name"
                onChange={updateLocalState}
              />
              <div id="lastnameError" className="error" />
            </div>
            <div>
              <input
                id="email"
                name="email"
                type="text"
                placeholder="Email"
                onChange={updateLocalState}
              />
              <div id="emailError" className="error" />
            </div>
            <div>
              <input
                id="phone"
                name="phoneNumber"
                type="tel"
                placeholder="Phone Number"
                onChange={updateLocalState}
              />
              <div id="phoneError" className="error" />
            </div>
            <div>
              <input
                id="passport"
                type="text"
                name="passportUrl"
                placeholder="PassportUrl"
                onChange={updateLocalState}
              />
              <div id="passportError" className="error" />
            </div>
            <div>
              <input
                id="password"
                name="password"
                type="password"
                placeholder="Password"
                onChange={updateLocalState}
              />
              <div id="password-error" className="error" />
            </div>
            <p>
              By creating an account you agree to our{' '}
              <a href="#">Terms & Privacy</a>.
            </p>
            <div className="form">
              <div id="signup-error" />
              <div id="signup-success" />
            </div>
            <div>
              <div className="loader" id="spinner" />
            </div>
            <div>
              <input id="submit-button" type="submit" value="SIGN UP" />
            </div>
            <div>
              <p>
                Already have an account? <a href="login.html">Log in Here</a>
              </p>
            </div>
          </form>
        </div>
      </div>
    </Layout>
  );
};

Signup.propTypes = {
  signupUser: PropTypes.func,
};

const mapStateToProps = state => ({
  isLoading: state.auth.isLoading,
  errors: state.auth.errors,
  isAuthenticated: state.auth.isAuthenticated,
});

export default connect(
  mapStateToProps,
  { signupUser },
)(Signup);
