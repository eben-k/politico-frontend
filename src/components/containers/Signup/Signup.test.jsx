import React from 'react';
import { shallow } from 'enzyme';

import { Signup } from './Signup';

const findByTestAttribute = (component, attribute) => {
  const wrapper = component.find(`[data-test='${attribute}']`);
  return wrapper;
};

const component = shallow(<Signup />);

describe('Custom Search', () => {
  xit('renders without errors', () => {
    expect(component).toMatchSnapshot();
  });

  xit('renders the form section', () => {
    expect(findByTestAttribute(component, 'reg-form').length).toBe(1);
  });
});
