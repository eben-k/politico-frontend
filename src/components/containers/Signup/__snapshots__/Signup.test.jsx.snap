// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Search renders without errors 1`] = `
<Layout>
  <div
    className="signup-form"
  >
    <h1
      className="login-heading"
    >
      Sign Up To Politico
    </h1>
    <div
      className="form"
    >
      <form
        action=""
        className="signup"
        id="form"
        onSubmit={[Function]}
      >
        <div>
          <input
            id="firstname"
            name="firstname"
            onChange={[Function]}
            placeholder="First Name"
            type="text"
          />
          <div
            className="error"
            id="firstnameError"
          />
        </div>
        <div>
          <input
            id="lastname"
            name="lastname"
            onChange={[Function]}
            placeholder="Last Name"
            type="text"
          />
          <div
            className="error"
            id="lastnameError"
          />
        </div>
        <div>
          <input
            id="email"
            name="email"
            onChange={[Function]}
            placeholder="Email"
            type="text"
          />
          <div
            className="error"
            id="emailError"
          />
        </div>
        <div>
          <input
            id="phone"
            name="phoneNumber"
            onChange={[Function]}
            placeholder="Phone Number"
            type="tel"
          />
          <div
            className="error"
            id="phoneError"
          />
        </div>
        <div>
          <input
            id="passport"
            name="passportUrl"
            onChange={[Function]}
            placeholder="PassportUrl"
            type="text"
          />
          <div
            className="error"
            id="passportError"
          />
        </div>
        <div>
          <input
            id="password"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
          />
          <div
            className="error"
            id="password-error"
          />
        </div>
        <p>
          By creating an account you agree to our
           
          <a
            href="#"
          >
            Terms & Privacy
          </a>
          .
        </p>
        <div
          className="form"
        >
          <div
            id="signup-error"
          />
          <div
            id="signup-success"
          />
        </div>
        <div>
          <div
            className="loader"
            id="spinner"
          />
        </div>
        <div>
          <input
            id="submit-button"
            type="submit"
            value="SIGN UP"
          />
        </div>
        <div>
          <p>
            Already have an account? 
            <a
              href="login.html"
            >
              Log in Here
            </a>
          </p>
        </div>
      </form>
    </div>
  </div>
</Layout>
`;
